src/datasource.ts

```diff
@@ -383,13 +383,13 @@ const recordSchemaArray = (name: string, types: AdxSchemaDefinition[], result: A
   if (types.every((t) => typeof t === 'string' && toPropertyType(t) === QueryEditorPropertyType.Number)) {
     // If all the types are numbers, the double takes precedence since it has more precission.
     const cslType = types.find((t) => typeof t === 'string' && (t === 'double' || t === 'real')) || defaultCslType;
-    result.push({ Name: name, CslType: cslType as string });
+    result.push({ Name: name, CslType: cslType as string, Type: 'dynamic' });
```

Comment:
I don't think we should use the `Type` field since that field is already a used field in a schema and it should be the [".NET equivalent type"](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/scalar-data-types/). I would instead extend the AdxColumnSchema type so we can inject some metadata into it:

```tsx
export interface AdxColumnSchema {
  Name: string;
  CslType: string;
  Type?: string;
  CslDefaultValue?: string;
  // tracking internally if a type is inferred from a dynamic type
  isDynamic?: boolean;
}
```

Reply:
Good shout 😊